lib/Inst/Inst.cpp

```diff
@@ -184,6 +184,9 @@ std::string ReplacementContext::printInstImpl(Inst *I, llvm::raw_ostream &Out,
           Out << " (powerOfTwo)";
         if (I->NumSignBits > 1)
           Out << " (signBits=" << I->NumSignBits << ")";
+        if (!I->Range.isEmptySet() && !I->Range.isFullSet())
```

Comment:
I think you should print the empty set case, unless it is somehow impossible

Reply:
you need to fix this-- LLVM can give us empty sets and it's wrong to not print that information that souper knows internally